src/app/data-model-provider/tests/TestActionReturnStatus.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "lib/core/CHIPError.h"
+#include "protocols/interaction_model/StatusCode.h"
+#include "pw_unit_test/framework_backend.h"
+#include <app/data-model-provider/ActionReturnStatus.h>
+#include <app/data-model-provider/StringBuilderAdapters.h>
+#include <lib/support/CodeUtils.h>
+
+#include <pw_unit_test/framework.h>
+
+using chip::app::DataModel::ActionReturnStatus;
+using chip::Protocols::InteractionModel::ClusterStatusCode;
+using chip::Protocols::InteractionModel::Status;
+
+TEST(TestActionReturnStatus, TestEquality)
+{
```

Comment:
What should the equivalence relation be for CHIP_NO_ERROR and ClusterStatusCode::ClusterSpecificSuccess?

Reply:
Added unit tests. This also got me to figure out equivalence between pure Status::Success and CHIP_NO_ERROR (which seem like they should be equivalent) so I fixed that. Thank you for the review!